cmd/coder/main.go

```diff
@@ -15,31 +15,35 @@ import (
 	"go.coder.com/flog"
 )
 
-var (
-	version string = "unknown"
-)
+// Using a global for the version so it can be set at build time using ldflags.
+var version string = "unknown"
 
 func main() {
 	ctx, cancel := context.WithCancel(context.Background())
 	defer cancel()
 
+	// If requested, spin up the pprof webserver.
 	if os.Getenv("PPROF") != "" {
 		go func() {
 			log.Println(http.ListenAndServe("localhost:6060", nil))
 		}()
 	}
 
+	// TODO: Document why this is needed. Only does something on windows.
 	stdoutState, err := xterminal.MakeOutputRaw(os.Stdout.Fd())
 	if err != nil {
-		flog.Fatal("set output to raw: %v", err)
+		flog.Fatal("set output to raw: %s", err)
 	}
-	defer xterminal.Restore(os.Stdout.Fd(), stdoutState)
+	defer func() {
+		// Best effort. Would result in broken terminal on window but nothing we can do about it.
+		_ = xterminal.Restore(os.Stdout.Fd(), stdoutState)
+	}()
 
 	app := cmd.Make()
 	app.Version = fmt.Sprintf("%s %s %s/%s", version, runtime.Version(), runtime.GOOS, runtime.GOARCH)
 
-	err = app.ExecuteContext(ctx)
-	if err != nil {
+	if err := app.ExecuteContext(ctx); err != nil {
+		// NOTE: The cmd lib will handle the logs, no need to do it here as well, simply exit with error code.
```

Comment:
what about this error? is it interesting?

Reply:
I'll reword the comment to be clearer. That error is already checked and log in the cmd lib. It is indeed an interesting one :)